main.c

```diff
@@ -407,12 +408,17 @@ static void update_info(void)
 	l->p = l->buf;
 	r->p = r->buf;
 	if (mode == MODE_THUMB) {
-		if (tns.loadnext < tns.end)
+		if (tns.loadnext < tns.end) {
 			bar_put(l, "Loading... %0*d", fw, tns.loadnext + 1);
-		else if (tns.initnext < filecnt)
+		} else if (tns.initnext < filecnt) {
 			bar_put(l, "Caching... %0*d", fw, tns.initnext + 1);
-		else
-			strncpy(l->buf, files[fileidx].name, l->size);
+		} else {
+			const char *filename = files[fileidx].name;
+			if ((homedir != NULL) && (strncmp(homedir, filename, strlen(homedir)) == 0 ))
+				bar_put(l, "~%s", (filename + strlen(homedir)));
+			else
+				strncpy(l->buf, filename, l->size);
```

Comment:
`strncpy` needs to be manually null terminated. But I see that there's a couple other places in the source where we're using strncpy without explicit null termination.

I'll fix them on a separate PR.

Reply:
Ok, have fixed but not pushed yet.